loadgen/loadgen.cc

```diff
@@ -345,16 +350,22 @@ std::vector<QueryMetadata> GenerateQueries(
 
   std::vector<QuerySampleIndex> samples(samples_per_query);
   std::chrono::nanoseconds timestamp(0);
+  //Choose a single sample to be used when in issue_same mode
+  QuerySampleIndex same_sample;
+  if (settings.issue_same) {
+     same_sample = sample_distribution_unique(sample_rng);
+  }
   while (timestamp <= k2xTargetDuration || queries.size() < min_queries) {
     if (kIsMultiStream) {
-      QuerySampleIndex sample_i = sample_distribution(sample_rng);
+      QuerySampleIndex sample_i = settings.issue_unique ? sample_distribution_unique(sample_rng) :
+                                                          sample_distribution(sample_rng);
       for (auto& s : samples) {
         // Select contiguous samples in the MultiStream scenario.
         // This will not overflow, since GenerateLoadableSets adds padding at
         // the end of the loadable sets in the MultiStream scenario.
         // The padding allows the starting samples to be the same for each
         // query as the value of samples_per_query increases.
-        s = loaded_samples[sample_i++];
+        s = (settings.issue_same) ? loaded_samples[same_sample] : loaded_samples[sample_i++];
```

Comment:
This will break the requirement that samples in a query should be contiguous for multistream.
For multistream, we should probably send the same query over and over.

To do that, revert this line and initialize samplie_i to same_sample every time.

Reply:
Since this feature will be solely used for auditing or test functionality - is it ok to break the requirement that samples in a query be contiguous for multistream. 
I ask because if sample_i is initalized to same_sample every time and spq = peformance size, then there will be no difference in performance_issue_unique vs performance_issue_same.

Reply:
> Since this feature will be solely used for auditing or test functionality - is it ok to break the requirement that samples in a query be contiguous for multistream.

We cannot break this requirement since some submitters have indicated their system will not work if the samples are not contiguous.

> I ask because if sample_i is initalized to same_sample every time and spq = peformance size, then there will be no difference in performance_issue_unique vs performance_issue_same.

This should be okay since it'll be different if spq != performance size.